Lesson4/src/com/home/model/fabric/Person/PersonFabric.java

```diff
@@ -16,13 +16,17 @@ public Person getSomeObject(List<String> initData) {
             return null;
         }
         String[] formattedData = initData.get(0).split(" ");
-        if (formattedData.length < 6) {
+        if (formattedData.length < 7) {
             System.err.println("Wrong input! Need more data!");
             return null;
         }
         try {
-            Person person = new Person(Integer.parseInt(formattedData[0]), formattedData[1], Integer.parseInt(formattedData[2]), formattedData[3].toCharArray()[0]);
-            Address address = new Address(formattedData[4], formattedData[5]);
+            Person person = new Person(Integer.parseInt(formattedData[0]),
+                    formattedData[1].toLowerCase().replaceFirst("^\\w", formattedData[1].substring(0, 1).toUpperCase()),
```

Comment:
this is ok, but Person has public constructor. So anyone can create a person with the wrong name or surname if they are not using your fabric

Reply:
fix